ml-proto/spec/check.ml

```diff
@@ -134,9 +134,9 @@ let rec check_expr c et e =
     check_expr_opt c (label c x) eo e.at
 
   | Br_if (x, eo, e) ->
+    check_expr_opt c et eo e.at;
```

Comment:
From a purely practical perspective, there's no problem with making

```
    (block (br_if $l (arg) (cond)) (other))
```

a type error, because can achieve a similar effect writing this code as

```
   (block (arg) (br_if $l (cond)) (other))
```

The practical restriction is that one can't use `br_if` to pass a value to one successor and drop it in the other. However, this restriction is consistent with how I expect to use `br_if`, and it's consistent with `br_if`'s operand being strict. It's evaluated for both values of the condition, so it makes sense to type-check it for both values of the condition.